tutorial/westeros/westeros.ipynb

```diff
@@ -0,0 +1,1268 @@
+    "        'year': horizon,\n",
+    "        'time': 'year',\n",
+    "        'value': demand_per_year * demand_profile,\n",
+    "        'unit': 'GWa',\n",
+    "    })\n",
+    "\n",
+    "scenario.add_par(\"demand\", light_demand)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "skip"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "#TODO Hide this one\n",
+    "\n",
+    "vintage_years, act_years = scenario.vintage_and_active_years()\n",
+    "\n",
+    "base_input = {\n",
+    "    'node_loc': country,\n",
+    "    'year_vtg': vintage_years,\n",
+    "    'year_act': act_years,\n",
+    "    'mode': 'standard',\n",
+    "    'node_origin': country,\n",
+    "    'commodity': 'electricity',\n",
+    "    'time': 'year',\n",
+    "    'time_origin': 'year',\n",
+    "}\n",
+    "\n",
+    "base_output = {\n",
+    "    'node_loc': country,\n",
+    "    'year_vtg': vintage_years,\n",
+    "    'year_act': act_years,\n",
+    "    'mode': 'standard',\n",
+    "    'node_dest': country,\n",
+    "    'time': 'year',\n",
+    "    'time_dest': 'year', \n",
+    "    'unit': '%',\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "Working backwards along the Reference Energy System, we can add connections for the `bulb`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "bulb_out = make_df(base_output, technology='bulb', commodity='light', level='useful', value=1.0)\n",
+    "scenario.add_par('output', bulb_out)\n",
+    "\n",
+    "bulb_in = make_df(base_input, technology='bulb', commodity='electricity',  level='final', value=1.0, unit='%')\n",
+    "scenario.add_par('input', bulb_in)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "Next, the `grid`, with loses of 13%"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "grid_efficiency = 0.87\n",
+    "grid_out = make_df(base_output, technology='grid', commodity='electricity', level='final', value=grid_efficiency)\n",
+    "scenario.add_par('output', grid_out)\n",
+    "\n",
+    "grid_in = make_df(base_input, technology='grid', commodity='electricity',  level='secondary', value=1.0, unit='%')\n",
+    "scenario.add_par('input', grid_in)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "And finally, our power plants which are not resource constrained in this simple model thus have no `input`"
```

Comment:
I wouldn't call this "not resource constrained", but "the model does not include the fossil resources used as inputs for coal_ppl". The costs of coal extraction are included in the variable costs." And wind obviously doesn't have an input, no need to phrase it in such a complicated way.

Reply:
done!